Fixes #77 - Added new nameof() tests to verify the fix for GMB-6506

gnysek opened the pull request:
Adds tests for `nameof()` (issue #77).

This is a test which could be helpful after fixing https://github.com/YoYoGames/GameMaker-Bugs/issues/6506 .

gnysek committed:
tests for nameof() - #77

gnysek committed:
added "var" test

gnysek committed:
reverse arguments, as "expected" is second

YYDan commented:
Please always file an issue describing the changes you want to add before adding a PR for undocumented changes.

gnysek commented:
This is for #77

YYDan commented:
Thanks - you beat me to it, but I was about to apologise and say I realised that the contribution guidelines in the Readme were _far_ too simplistic and so I have updated them now to be correct ;)

gnysek commented:
Yeah, there was no info about it, that's why I didn't added issue in first place :) But no problem!

gnysek committed:
Merge branch 'YoYoGames:develop' into develop

AdamBorek commented on projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:
```diff
@@ -0,0 +1,33 @@
+global.NAMEOF_GLOBAL_TEST_STRING = "foobar";
+
+function BasicNameofTestSuite() : TestSuite() constructor {
+	
```
```suggestion

	// Test names of macros
```

AdamBorek commented on projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:
```diff
@@ -0,0 +1,33 @@
+		assert_equals(nameof(NAMEOF_TEST_REAL), "NAMEOF_TEST_REAL", "#1 nameof of macro which holds real value");
+		assert_equals(nameof(NAMEOF_TEST_STRING), "NAMEOF_TEST_STRING", "#2 nameof of macro which holds string value");
+	});
+	
```
```suggestion
	
	// Test names of enums
```

AdamBorek commented on projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:
```diff
@@ -0,0 +1,33 @@
+		assert_equals(nameof(nameof_test.entry_1), "nameof_test.entry_1", "#3 nameof of enum which holds real value");
+		assert_equals(nameof(nameof_test.entry_2), "nameof_test.entry_2", "#4 nameof of enum which holds string value");
+	});
+	
```
```suggestion
	
	// Test names of global variables
```

AdamBorek commented on projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:
```diff
@@ -0,0 +1,33 @@
+		assert_equals(nameof(global.NAMEOF_GLOBAL_TEST_REAL), "global.NAMEOF_GLOBAL_TEST_REAL", "#5 nameof of global which holds real value");
+		assert_equals(nameof(global.NAMEOF_GLOBAL_TEST_STRING), "global.NAMEOF_GLOBAL_TEST_STRING", "#6 nameof of global which holds string value");
+	});
+	
```
```suggestion
	
	// Test names of local variables
```

AdamBorek commented on projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:
```diff
@@ -0,0 +1,33 @@
+#macro NAMEOF_TEST_REAL 21.37
```
```suggestion
// Add variables to test names of
#macro NAMEOF_TEST_REAL 21.37
```

YYDan committed:
Update projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml

Co-authored-by: AdamBorek <aborek@opera.com>

YYDan committed:
Update projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml

Co-authored-by: AdamBorek <aborek@opera.com>

YYDan committed:
Update projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml

Co-authored-by: AdamBorek <aborek@opera.com>

YYDan committed:
Update projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml

Co-authored-by: AdamBorek <aborek@opera.com>

YYDan committed:
Update projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml

Co-authored-by: AdamBorek <aborek@opera.com>

AdamBorek reviewed (commented):
All looks good, just added some comments above the facts and declarations

YYDan committed:
Merge branch 'develop' into develop

gnysek committed:
Update BasicNameofTestSuite.gml - tab character test for macros

Added test for regression I found in 2024.800 while checking fix for https://github.com/YoYoGames/GameMaker-Bugs/issues/6506, where if there's Tab (`\t`) character in front of `#macro`, `nameof()` returns empty string (``).

gnysek commented:
I've added another test, as during checking Squirrel (`-I`) version of betas, I found an interesting bug, that if there's `\t` in front of Tab, macro would be converted to empty string in case of `nameof()`. It's then worth to check it too in future tests, to prevent regression.

YYDan commented:
Thanks - although, if adding more tests later on, ensure you add a separate new issue and PR for just _that bit_ rather than adding it later on into an existing PR awaiting review ;)

DiasFranciscoA commented:
@gnysek apparently there are some conflicts with the dest branch, please review this PR so we can merge it

gnysek committed:
Merge branch 'develop' into develop

gnysek committed:
updated `.resource_order`

gnysek commented:
@DiasFranciscoA I'm not sure if `.resource_order` file should be in this repo. It could cause problems every time that during one pull request is made, another one with another new resource(s) is merged.
I've fixed it so merged file contains all resources, then reopened project in GM, to ensure .resource_order is up to date.

DiasFranciscoA commented:
Hmm that makes sense, I'll make sure `.resource_order` is removed from the repo and added to `.gitignore`

DiasFranciscoA commented:
@gnysek can you please resolve the conflicts the `.resource_order` file is now out of the picture.

EDITED: Fixed locally and pushed to develop (thank you for the contribution)
